Public/Src/Engine/Processes/SandboxedProcessReports.cs

```diff
@@ -522,7 +529,77 @@ private void AddLookupEntryForProcessExit(uint processId, ReportedProcess report
             m_processesExits[processId] = reportedProcess;
         }
 
-        private bool FileAccessReportLineReceived<T>(ref T data, FileAccessReportProvider<T> parser, out string errorMessage)
+        private bool TryParseAugmentedFileAccess(
+            ref string data,
+            out uint processId,
+            out ReportedFileOperation operation,
+            out RequestedAccess requestedAccess,
+            out FileAccessStatus status,
+            out bool explicitlyReported,
+            out uint error,
+            out Usn usn,
+            out DesiredAccess desiredAccess,
+            out ShareMode shareMode,
+            out CreationDisposition creationDisposition,
+            out FlagsAndAttributes flagsAndAttributes,
+            out AbsolutePath manifestPath,
+            out string path,
+            out string enumeratePattern,
+            out string processArgs,
+            out string errorMessage)
+        {
+            // An augmented file access has the same structure as a regular one, so let's call
+            // the usual parser
+            var result = FileAccessReportLine.TryParse(
+                ref data, 
+                out processId, 
+                out operation, 
+                out requestedAccess, 
+                out status, 
+                out explicitlyReported, 
+                out error, 
+                out usn, 
+                out desiredAccess, 
+                out shareMode, 
+                out creationDisposition, 
+                out flagsAndAttributes, 
+                out manifestPath, 
+                out path, 
+                out enumeratePattern, 
+                out processArgs, 
+                out errorMessage);
+
+            // Augmented file accesses never have the manifest path set, since there is no easy access to the manifest for 
+            // processes to use
+            // Let's recreate the manifest path based on the current path and manifest
+            if (string.IsNullOrEmpty(path) || !AbsolutePath.TryCreate(m_pathTable, path, out var absolutePath))
+            {
+                return result;
+            }
+            
+            var success = m_manifest.TryFindManifestPathFor(absolutePath, out AbsolutePath computedManifestPath, out FileAccessPolicy policy);

```

Comment:
>m_manifest.TryFindManifestPathFor [](start = 26, length = 33)

This unfortunately won't work on VM because the manifest is in the form of array of bytes instead of a tree. 
Either
- Rehydrate the tree
- Have the same path. Please add comment for the reason. Better, add a unit test. There's an integration unit test for this: `Public\Src\Engine\UnitTests\Scheduler.IntegrationTest\ExternalToolExecutionTests.cs`

Reply:
Went with the first option. Will add a test.

---
In reply to: [343856730](https://github.com/microsoft/BuildXL/pull/1175#discussion_r343856730) [](ancestors = 343856730)